demos/jans-tarp/src/popup.js

```diff
@@ -419,4 +429,43 @@ const qs = require('qs');
     );
   }
 
+  async function generateRandomChallengePair() {
```

Comment:
No need to create seperate variable `randString` to store `generateRandomString()`. Directly assign the function to `secret` and remove `await` like @mjatin-dev mentioned because the functions are not `Promise` based.